service/service_windows.go

```diff
@@ -0,0 +1,135 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package service
+
+import (
+	"fmt"
+	"log"
+	"syscall"
+
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+type WindowsService struct {
+	app    *Application
+	params Parameters
+}
+
+func NewWindowsService(params Parameters) *WindowsService {
+	return &WindowsService{params: params}
+}
+
+func (s *WindowsService) Execute(args []string, requests <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	if len(args) == 0 {
+		log.Fatal("expected first argument supplied to service.Execute to be the service name")
+	}
+
+	changes <- svc.Status{State: svc.StartPending}
+	s.start(args[0], s.params)
+	changes <- svc.Status{State: svc.Running, Accepts: svc.AcceptStop | svc.AcceptShutdown}
+
+	for req := range requests {
+		switch req.Cmd {
+		case svc.Interrogate:
+			changes <- req.CurrentStatus
+		case svc.Stop, svc.Shutdown:
+			changes <- svc.Status{State: svc.StopPending}
+			s.stop()
+			changes <- svc.Status{State: svc.Stopped}
+			return
+		default:
+			log.Fatalf(fmt.Sprintf("unexpected control request #%d", req))
```

Comment:
`%d` is for integers, while `req` if of type `svc.ChangeRequest` (struct), use `%v` instead. See https://gobyexample.com/string-formatting
Also, propagate error.

Reply:
Oops, nice catch - this was meant to be `req.Cmd`